### ISSUE TITLE
Loss spikes and explode with mpt-1b model pretrain

### ISSUE THREAD

**Issue by sagorbrur** [mosaicml/llm-foundry#759]
Hi,
We are training an MPT-1B model with 230 GB of Bi-lingual (Bangla + English) data.
We keep the default config with changes in batch size and start training in the H100 (8GPU) machine.
Loss was decreasing but after 42k iteration it got spiked and started exploding. (Attachment)
![image](https://github.com/mosaicml/llm-foundry/assets/10723655/6956093b-210b-442b-abed-d8dc19c73d57)

We tried different tuning like
- Stop and resume training
- Reduce LR and increase warmup from 100 to 1000
- Change weight decay from 0.0 to 0.1
- Load only weight and start with a small LR

None of the above methods is mitigating our exploding problem. 
If anyone has any suggestions to mitigate this problem please let me know.

**Comment by vchiley** [comment]
https://github.com/mosaicml/llm-foundry/issues/232#issuecomment-1565653341
